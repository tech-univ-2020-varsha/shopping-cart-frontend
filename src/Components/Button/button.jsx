import React from 'react';
import propTypes from 'prop-types';


const Button = ({ buttonText, clickEvent, testid }) => (
  <button
    type="submit"
    className="buttonStyle"
    onClick={clickEvent}
    data-testid={testid}
  >
    {buttonText}
  </button>
);


Button.defaultProps = {
  testid: 'test-id',
};
Button.propTypes = {
  buttonText: propTypes.string.isRequired,
  clickEvent: propTypes.func.isRequired,
  testid: propTypes.string,
};
export default Button;
