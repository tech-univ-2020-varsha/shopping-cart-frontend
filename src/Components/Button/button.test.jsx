import { render, fireEvent } from '@testing-library/react';
import React from 'react';
import Button from './button';

describe('the button component', () => {
  it('should render correctly', () => {
    const click = () => {
    };
    const { asFragment } = render(<Button
      buttonText="text"
      clickEvent={click}
      testid="btn-test-id"
    />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should call the click function on click event', () => {
    const click = jest.fn();
    const { getByTestId } = render(<Button
      buttonText="text"
      clickEvent={click}
      testid="btn-test-id"
    />);

    fireEvent.click(getByTestId('btn-test-id'));
    expect(click).toHaveBeenCalled();
  });
});
