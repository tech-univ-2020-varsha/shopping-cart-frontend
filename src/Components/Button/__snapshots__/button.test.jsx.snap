// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the button component should render correctly 1`] = `
<DocumentFragment>
  <button
    data-testid="btn-test-id"
    type="submit"
  >
    text
  </button>
</DocumentFragment>
`;
